src/framework/textFilters.js

```diff
@@ -0,0 +1,255 @@
+        this.top.push(String.fromCharCode(856));
+        this.bottom.push(String.fromCharCode(857));
+        this.bottom.push(String.fromCharCode(858));
+        this.top.push(String.fromCharCode(859));
+        this.bottom.push(String.fromCharCode(860));
+        this.top.push(String.fromCharCode(861));
+        this.top.push(String.fromCharCode(861));
+        this.bottom.push(String.fromCharCode(863));
+        this.top.push(String.fromCharCode(864));
+        this.top.push(String.fromCharCode(865));
+    },
+};
+ZALGO_DIACRITICS.init();
+
+function normal(text) {
+    return text;
+}
+
+function flipped(text) {
+    return [...text].map(ch => FLIP_MAP[ch] || ch)
+        .reverse()
+        .join('');
+}
+
+function bubbles(text) {
+    // Circles around Letters. Uses special circle characters for some letters
+    // and combining characters for the rest
+    // Thanks to
+    // - Alan Wood: http://www.alanwood.net/unicode/enclosed_alphanumerics.html
+    let first = true;
+
+    const ret = [...text].map(letter => {
+        let ch = BUBBLES_MAP[letter];
+
+        // No dedicated circled character available? Use a Combining Diacritical
+        // Mark surrounded with non-breaking spaces, so it doesn't overlap
+        if (typeof ch === 'undefined') {
+            if (letter.charCodeAt(0) >= 33) {
+                ch = letter + String.fromCharCode(8413);
+                if (!first) {
+                    ch = String.fromCharCode(8239) + String.fromCharCode(160) +
+                        String.fromCharCode(160) + String.fromCharCode(8239) +
+                        ch;
+                }
+            } else {
+                ch = letter;
+            }
+        }
+        first = ch === '\n';
+        return ch;
+    });
+    return ret.join('');
+}
+
+function scrambled(text) {
+    let words = text.split(' ');
+    words = words.map(word => {
+        const sequence = Array.from({length: word.length - 2}, () => Math.random());
+        const letters = word.split('');
+        const beginning = letters[0];  // eslint-disable-line prefer-destructuring
+        const middle = Utils.shuffle(letters.slice(1, -1), sequence);
+        const end = letters[letters.length - 1];
+        return beginning + middle.join('') + end;
+    });
+    return words.join(' ');
+}
+
+function zalgo(text) {
+    // Thanks to Michael S. Kaplan:
+    // http://blogs.msdn.com/b/michkap/archive/2006/02/17/533929.aspx
+
+    // How many diacritic marks shall we put on top/bottom?
+    const MAX_HEIGHT = 15;
+    // 0-100%. maxHeight 100 and randomization 20%: the height goes from 80 to
+    // 100. randomization 70%, height goes from 30 to 100.
+    const RANDOMIZATION = 100;
+
+    let newText = '';
+    [...text].forEach(letter => {
+        let newChar = letter;
+
+        // Middle
+        // Put just one of the middle characters there, or it gets crowded
+        const middleLen = ZALGO_DIACRITICS.middle.length;
```

Comment:
Should this not be length -1?

Reply:
No, I don't think so - `Math.floor(Math.random() * middleLen)` will return an index from 0 to `middleLen - 1`, so it works out okay.